### ISSUE TITLE
zoekt: add structured logging for git fetch duration and indexing duration metrics

### ISSUE THREAD

**Issue by ggilmore** [sourcegraph/sourcegraph#34951]
We're in the process of gathering metrics to examine the performance benefits of incremental indexing v. the old indexing strategy. We currently use Prometheus and Grafana to collect, aggregate, and display this data. However in this scenario, the bucketing that Prometheus uses produces charts that are both misleading and unhelpful. 

See this [Slack thread for more context](https://sourcegraph.slack.com/archives/C023ELQLV7F/p1651608207462129). 

Scraping the logs directly (as described in the slack thread) provides us with more accurate data that we can manually create graphs from. However, as as it stands the logs for `git fetch` duration and indexing times are unstructured and hard to robustly parse (with something like `awk`). 

It would be nice to convert these logs use a structured logger (like https://github.com/uber-go/zap) instead so that we can more easily parse them.

Goals:

- [ ] produce structured log output for the `git fetch` + total indexing time logs
- [ ] create an example `awk` script that we could use to process these logs and pipe to a charting tool 

Non-goals:

- convert all of zoekt to use structured logging: It would be amazing of all of our logs were structured and easy to parse, but doing that is far beyond the scope of this issue. Once this lands, we can incrementally work towards converting all of the preexisting unstructured logs to use this new method,

**Comment by gl-srgr** [comment]
Another slack thread that's relevant (this conversation has an older start date than the slack thread posted in the Issue description): https://sourcegraph.slack.com/archives/C023ELQLV7F/p1647435239044179

**Comment by gl-srgr** [comment]
resolved with: 
Structured logging for indexserver using sourcegraph/log sourcegraph/zoekt#374
